src/components/obb-clipping.js

```diff
@@ -0,0 +1,402 @@
+/* global AFRAME */
+
+AFRAME.registerComponent('obb-clipping', {
+  schema: {
+    size: { default: 0 },
+    clippingSourceSelectorString: {
+      default: '[street]' // we use string instead of selector as the built-in a-frame selector doesn't support this syntax
+    },
+    clippingDestinationSelectorString: {
+      // empty string means use the same element as the source
+      type: 'string'
+    },
+    minimumColliderDimension: { default: 0.02 }
+  },
+
+  init: function () {
+    this.tick = AFRAME.utils.throttleTick(this.tick, 250, this);
```

Comment:
for my education, what's the reason behind this?

Reply:
this throttles tick to only fire every 250ms instead of every frame. recalculating the bounding box and then traversing all the tiles to apply the clipping planes is very expensive.

better solution is to monitor for pose changes on target object OR tiles loaded events from 3dtiles runtime object via threejs-loader-tiles library and ONLY THEN initiate updating obb and clipping planes, but throttling is a cheap easy way to do it for now